alias/src/main/java/software/amazon/kms/alias/AliasHelper.java

```diff
@@ -0,0 +1,111 @@
+        return wrapKmsExceptions(CREATE_ALIAS,
+            () -> proxyClient.injectCredentialsAndInvokeV2(createAliasRequest,
+                proxyClient.client()::createAlias));
+    }
+
+    public DeleteAliasResponse deleteAlias(final DeleteAliasRequest deleteAliasRequest,
+                                           final ProxyClient<KmsClient> proxyClient) {
+        return wrapKmsExceptions(DELETE_ALIAS,
+            () -> proxyClient.injectCredentialsAndInvokeV2(deleteAliasRequest,
+                proxyClient.client()::deleteAlias));
+    }
+
+    public ListAliasesResponse listAliases(final ListAliasesRequest deleteAliasRequest,
+                                           final ProxyClient<KmsClient> proxyClient) {
+        return wrapKmsExceptions(LIST_ALIASES,
+            () -> proxyClient.injectCredentialsAndInvokeV2(deleteAliasRequest,
+                proxyClient.client()::listAliases));
+    }
+
+    public UpdateAliasResponse updateAlias(final UpdateAliasRequest updateAliasRequest,
+                                           final ProxyClient<KmsClient> proxyClient) {
+        return wrapKmsExceptions(UPDATE_ALIAS,
+            () -> proxyClient.injectCredentialsAndInvokeV2(updateAliasRequest,
+                proxyClient.client()::updateAlias));
+    }
+
+    private <T> T wrapKmsExceptions(final String operation, final Supplier<T> serviceCall) {
+        try {
+            return serviceCall.get();
+        } catch (final AlreadyExistsException e) {
+            throw new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, e.getMessage());
+        } catch (final InvalidAliasNameException | KmsInvalidStateException | InvalidArnException e) {
+            throw new CfnInvalidRequestException(e);
+        } catch (final LimitExceededException e) {
+            throw new CfnServiceLimitExceededException(ResourceModel.TYPE_NAME, e.getMessage());
+        } catch (final InvalidMarkerException e) {
+            // We should never make a call with an invalid marker, if we did, there is an issue
+            throw new CfnInternalFailureException(e);
+        } catch (final KmsInternalException | DependencyTimeoutException e) {
+            throw new CfnServiceInternalErrorException(e);
+        } catch (final NotFoundException e) {
+            throw new CfnNotFoundException(e);
+        } catch (final KmsException e) {
+            if (ACCESS_DENIED_ERROR_CODE.equals(e.awsErrorDetails().errorCode())) {
+                throw new CfnAccessDeniedException(operation, e);
+            } else if (VALIDATION_ERROR_CODE.equals(e.awsErrorDetails().errorCode())) {
+                throw new CfnInvalidRequestException(e);
+            }
+
+            throw new CfnGeneralServiceException(operation, e);
+        } catch (final AmazonServiceException exception) {
+            if (THROTTLING_ERROR_CODE.equals(exception.getErrorCode())) {
+                throw new CfnThrottlingException(operation, exception);
+            }
```

Comment:
Ah good catch. I called KMS myself until I got throttled and got the following exception:
```
software.amazon.awssdk.services.kms.model.KmsException: You have exceeded the rate at which you may call KMS. Reduce the frequency of your calls. (Service: Kms, Status Code: 400, Request ID: 0e757526-9bdf-4500-954c-f9eea58c3d93, Extended Request ID: null)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleErrorResponse(CombinedResponseHandler.java:123)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleResponse(CombinedResponseHandler.java:79)
...
```

Will update so that we are catching the correct exception.